server/images.go

```diff
@@ -427,12 +430,37 @@ func CreateModel(ctx context.Context, name string, path string, fn func(resp api
 	// Create a single layer for the parameters
 	if len(params) > 0 {
 		fn(api.ProgressResponse{Status: "creating parameter layer"})
-		layers = removeLayerFromLayers(layers, "application/vnd.ollama.image.params")
+
 		formattedParams, err := formatParams(params)
 		if err != nil {
 			return fmt.Errorf("couldn't create params json: %v", err)
 		}
 
+		paramsLayer, newLayers := removeLayerFromLayers(layers, "application/vnd.ollama.image.params")
```

Comment:
Looks like this is reading the params layer from the current modelfile rather than the parent?

I tried this with a modelfile that doesn't set any parameters and it didnt get any parameters set from the parent since `len(params) == 0`

Reply:
Good call. I'll move it out so it's unconditional

Reply:
This is addressed in the latest update